test/functional/legacy/061_undo_tree_spec.lua

```diff
@@ -0,0 +1,256 @@
+-- Tests for undo tree and :eralier and :later.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source, eq, eval, clear, execute, expect, wait =
+  helpers.feed, helpers.insert, helpers.source, helpers.eq, helpers.eval,
+  helpers.clear, helpers.execute, helpers.expect, helpers.wait
+
+local expect_empty_buffer = function()
```

Comment:
Note that we prefer the `local function foo()` syntax instead of `local foo = function()`. (Mostly because we just need to pick one and stick with it)